lib/logic/task_instance/task_instance_cubit.dart

```diff
@@ -0,0 +1,159 @@
+			taskInstance.duration.add(DateSpan(from: TimeDate.now()));
+			updates.add(_dataRepository.updateTaskInstance(taskInstance));
+
+			if(planInstance.duration == null) planInstance.duration = [];
+			planInstance.duration.add(DateSpan(from: TimeDate.now()));
+			wasPlanInstanceChanged = true;
+		}
+		if(wasPlanInstanceChanged) toUpdate.add(planInstance);
+		updates.add(_dataRepository.updateMultiplePlanInstances(toUpdate));
+		await Future.value(updates);
+
+		Task task = await _dataRepository.getTask(taskId: taskInstance.taskID);
+		UITaskInstance uiTaskInstance = UITaskInstance.singleFromDBModel(task: taskInstance, name: task.name, description: task.description, points: task.points != null ? UIPoints(quantity: task.points.quantity, title: task.points.name, createdBy: task.points.createdBy, type: task.points.icon) : null);
+		_activeTaskService.setTaskStateActive();
+		if(isInProgress(uiTaskInstance.duration)) emit(TaskInstanceStateProgress(uiTaskInstance,  await _getUiPlanInstance(taskInstance.planInstanceID)));
+		else  emit(TaskInstanceStateBreak(uiTaskInstance,  await _getUiPlanInstance(taskInstance.planInstanceID)));
+	}
+
+	void switchToBreak() async {
+		TaskInstance taskInstance = await _dataRepository.getTaskInstance(taskInstanceId: _taskInstanceId);
+		taskInstance.duration.last.to = TimeDate.now();
+		taskInstance.breaks.add(DateSpan(from: TimeDate.now()));
+		await _dataRepository.updateTaskInstance(taskInstance);
+		Task task = await _dataRepository.getTask(taskId: taskInstance.taskID);
+		UITaskInstance uiTaskInstance = UITaskInstance.singleFromDBModel(task: taskInstance, name: task.name, description: task.description, points: task.points != null ? UIPoints(quantity: task.points.quantity, title: task.points.name, createdBy: task.points.createdBy, type: task.points.icon) : null);
+		emit(TaskInstanceStateBreak(uiTaskInstance,  await _getUiPlanInstance(taskInstance.planInstanceID)));
+	}
+
+	void switchToProgress() async {
+		TaskInstance taskInstance = await _dataRepository.getTaskInstance(taskInstanceId: _taskInstanceId);
+		taskInstance.breaks.last.to = TimeDate.now();
+		taskInstance.duration.add(DateSpan(from: TimeDate.now()));
+		await _dataRepository.updateTaskInstance(taskInstance);
+		Task task = await _dataRepository.getTask(taskId: taskInstance.taskID);
+		UITaskInstance uiTaskInstance = UITaskInstance.singleFromDBModel(task: taskInstance, name: task.name, description: task.description, points: task.points != null ? UIPoints(quantity: task.points.quantity, title: task.points.name, createdBy: task.points.createdBy, type: task.points.icon) : null);
+		emit(TaskInstanceStateProgress(uiTaskInstance, await _getUiPlanInstance(taskInstance.planInstanceID)));
+	}
+
+	void markAsDone() async {
+		TaskInstance taskInstance = await _dataRepository.getTaskInstance(taskInstanceId: _taskInstanceId);
+		PlanInstance planInstance = await _dataRepository.getPlanInstance(id: taskInstance.planInstanceID);
```

Comment:
Same with the `PlanInstance` - keep and update a local model instead to save time